src/OpenTelemetry.Instrumentation.AspNet.Win/Implementation/HttpInListener.cs

```diff
@@ -59,23 +61,38 @@ public override void OnStartActivity(Activity activity, object payload)
                 return;
             }
 
-            // TODO: Avoid the reflection hack once .NET ships new Activity with Kind settable.
-            activity.GetType().GetProperty("Kind").SetValue(activity, ActivityKind.Server);
-
             var request = context.Request;
             var requestValues = request.Unvalidated;
 
+            if (!(this.options.TextFormat is TraceContextFormatActivity))
+            {
+                // This requires to ignore the current activity and create a new one
+                // using the context extracted using the format TextFormat supports.
+                var ctx = this.options.TextFormat.Extract<HttpRequest>(
+                    request,
+                    (r, name) => requestValues.Headers.GetValues(name));
+
+                Activity newOne = new Activity(ActivityNameByHttpInListener);
+                newOne.SetParentId(ctx.TraceId, ctx.SpanId, ctx.TraceFlags);
+                newOne.TraceStateString = ctx.TraceState;
+                newOne.Start();
+                activity = newOne;
+            }
+
             // see the spec https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/data-semantic-conventions.md
             var path = requestValues.Path;
             activity.DisplayName = path;
 
+            // TODO: Avoid the reflection hack once .NET ships new Activity with Kind settable.
+            activity.GetType().GetProperty("Kind").SetValue(activity, ActivityKind.Server);
+
             var samplingParameters = new ActivitySamplingParameters(
-                activity.Context,
-                activity.TraceId,
-                activity.DisplayName,
-                activity.Kind,
-                activity.Tags,
-                activity.Links);
+                        activity.Context,
```

Comment:
nit: Extra tab thing going on here? Also at the end of the method (line 121) spacing looks off. Sorry for even mentioning it just happened to catch my eye.